torch/autograd/_functions/tensor.py

```diff
@@ -591,6 +591,16 @@ def backward(self, grad_output):
         return grad_input
 
 
+def reverse_cumsum(x, dim):
+    ret = torch.cumsum(-x, dim=dim)
+
+    end_idx = ret.size(dim) - 1
+    ret_sum = ret.narrow(dim, end_idx, 1)
```

Comment:
Can you add a `clone()` here? I just realized that `ret` and `ret_sum` share storage, so `-=` doesn't necessarily has to be valid (apparently it is at the moment, but we don't guarantee it).